server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -142,6 +142,15 @@ func (s *BuildBuddyServer) DeleteInvocation(ctx context.Context, req *inpb.Delet
 	return &inpb.DeleteInvocationResponse{}, nil
 }
 
+func (s *BuildBuddyServer) CancelInvocation(ctx context.Context, req *inpb.CancelInvocationRequest) (*inpb.CancelInvocationResponse, error) {
+	res := s.env.GetRemoteExecutionService()
```

Comment:
we'll want a perms check here to check that the user has write access to the invocation, since it looks like execution_server doesn't check those permissions for us. The write access check is important because workflow invocations can be public, but we don't want any old user to be able to cancel a workflow invocation.

You could use env.GetInvocationDB() for looking up the invocation, which checks read access for the authenticated user. Then you could use perms.AuthorizeWrite() to check write access as well.

Reply:
Done